backend/data_validation/base_data_val.py

```diff
@@ -29,25 +32,79 @@ def __init__(self, path_to_config: str = "config.yaml"):
-            ) as file:
-                self.match_schedule = load(file)
-        except FileNotFoundError:  # We want to ignore if it doesn't exist because get_match_schedule() will create it.
-            pass
+        self._event_key = str(self.config["year"]) + self.config["event_code"]
+        # determines both if were using tba for match shedule and whether we're running tba checks
+        self._run_with_tba = self.config.get("run_with_tba", True)
 
-        # Setting up FalconAlliance (our connection to TBA) and retrieving match schedule if it doesn't exist
+        # Setting up FalconAlliance (our connection to TBA) and retrieving match schedule
         self.api_client = falcon_alliance.ApiClient(
             api_key="6lcmneN5bBDYpC47FolBxp2RZa4AbQCVpmKMSKw9x9btKt7da5yMzVamJYk0XDBm"  # for testing purposes
         )
 
-        self._event_key = str(self.config["year"]) + self.config["event_code"]
-        self._run_tba_checks = self.config.get("run_tba_checks", True)
+        with self.api_client:
+            # make sure tba isn't down
+            self._run_with_tba = (
+                self._event_key not in self.api_client.status().down_events
+            )
+
+            if self._run_with_tba:
+                self.get_match_schedule_tba()
+            else:
+                self.get_match_schedule_file()
+
+    @abstractmethod
+    def validate_data(self, scouting_data: list = None) -> None:
+        """
+        Runs all checks validating a single submission from 2022's game (Rapid React).
+
+        :param scouting_data: Optional parameter containing scouting data mostly for testing purposes.
+        :return:
+        """
+        pass
+
+    @abstractmethod
+    def validate_submission(self, submission: Series) -> None:
+        """
+        Runs all checks validating a single submission.
+
+        :param submission: Series object containing a single submission of scouting data.
+        :return:
+        """
+        pass
+
+    def check_team_info_with_match_schedule(
+        self,
+        match_key: str,
+        team_number: int,
+        alliance: str,
+        driver_station: int,
+    ) -> None:
+        """
+        Checks if the team scouted was in the match on given alliance
+        Checks if the team driverstation doesn't correspond to team scouted
+
+        :param match_key: Key of match that was scouted.
+        :param team_number: Number of team that was scouted (eg 4099).
+        :param alliance: string either "Red' or 'Blue' of which alliance was scouted
+        :param driver_station: int representing which driver station(1-3) was  scouted.
+        """
+        match_schedule_key = self._event_key + "_" + match_key
+        teams_on_alliance = self.match_schedule[match_schedule_key][alliance]
+        teams_on_alliance = list(
+            map(lambda team: int(team.lstrip("frc")), teams_on_alliance)
+        )
```

Comment:
We could use a list comprehension here to keep things more concise, but it doesn't really matter at the end of the day so you don't need to accept this suggestion
```suggestion
        teams_on_alliance = [int(team.lstrip("frc")) for team in teams_on_alliance]
```